dvc/repo/ls.py

```diff
@@ -0,0 +1,73 @@
+import os
+
+from dvc.exceptions import DvcException, OutputNotFoundError
+
+
+@staticmethod
+def ls(url, target=None, rev=None, recursive=None, outputs_only=False):
+    from dvc.external_repo import external_repo
+    from dvc.repo import Repo
+    from dvc.utils import relpath
+
+    with external_repo(url, rev) as repo:
+        result = []
```

Comment:
Why not `set()` right away? You are converting it later anyway

Reply:
Cause I am everywhere uses list (as outs from `_ls_outs_repo`  and `_ls_files_repo`) and expect to return sorted output. I'm using `set` only for skipping possible duplicates, but not as a general data-structure